### DISCUSSION TITLE
why settings from JSON file don't overrides default values of settings defined in pydantic model?

### DISCUSSION THREAD

**Discussion by yashrathi-git** [pydantic/pydantic#3017] [Question] [answered]
Hi, I want to store all my application settings in a JSON, file and want pydantic to read and write to it. Does pydantic offer a convenient way to do that, or I will have to do it myself? I have to reference these settings in multiple places throughout my application. It would be great if someone could help on what would be the best way to do so.

Currently, I have this code

```py
import json
from typing import Dict, Any

from pydantic import BaseSettings, BaseModel


def json_config_settings_source(settings: BaseSettings) -> Dict[str, Any]:
    with open("path/to/json/file.json") as file:
        contents = json.load(file)
    return contents


class SubSetting(BaseModel):
    setting_1: str='value'
    setting_2: str='value'


class Defaults(BaseModel):
    colors: SubSetting = SubSetting()


class Settings(Defaults):

    class Config:
        @classmethod
        def customise_sources(
                cls,
                init_settings,
                env_settings,
                file_secret_settings,
        ):
            return json_config_settings_source,
```
No matter what I change in my JSON file, this `Settings` class instance always returns the default values? Why is that and what would be the best way to implement something like this?

**Comment by lesnik512** [comment, marked as answer]
Maybe the problem is that `Settings` should be a subclass of `BaseSettings`?

**Reply by yashrathi-git** [reply]
Yes, that was the problem. Thank you so much, was debugging it for a long time, and for some reason didn't notice that. Does pydantic also offer a way to save data back to file or do I make a `save` method?

**Reply by lesnik512** [reply]
You can call `.json()` method on object to receive a json-string and than save it to a file with Python's tools.

BTW, there is a way to parse json file with subclasses of BaseModel also.
You can use `parse_file` method. Here is the [link](https://pydantic-docs.helpmanual.io/usage/models/#helper-functions)
